### ISSUE TITLE
Is configure still supported in 2.3.1 ?

### ISSUE THREAD

**Issue by zarniwhoop73** [ice-wm/icewm#46]
As a long-time user (only on linux), I find the vagaries of using cmake across multiple projects annoying. But I have seen suggestions that you now recommend cmake (although the shipped configure.sh still uses configure).

Before I ask about enabling libpng and libjpeg in configure (I can see the tests in the script generated from autogen, but they do not appear to be run) I thought I had better ask if configure is still supported :)

**Comment by rubyFeedback** [comment]
Not sure what you mean with "supported". If you extract the tarball e. g. of 2.4.0 right now, and do "./configure", you get the configure run, yes? A CMakeLists.txt also exists. So that means both configure and cmake work right now.

If your comment is about support for libpng and libjpeg, I am not sure if e. g. libpng is needed; --enable-libpng does not seem to work in GNU configure. Is that your request to have this supported in configure? I assume gijsbers would not reject patches that add support for this into the GNU configure macro thingies.

On a side note: I find the GNU configure system awful, pre-2000 with a horrible syntax. cmake made this a bit better; meson/ninja are IMO even better. It's unfortunate that there is a proliferation of build tools, and we can cite xkcd for added fun   https://xkcd.com/927/ - but there were reasons why e. g. the KDE team moved away from GNU configure. It's not that new build tools are always better either, but "old and existing" can be immensely annoying as well. Personally I'd never use GNU configure for new projects myself, unless there is some specific overarching need to do so. (I compile stuff via ruby in an automated manner; either with hardcoding which build system is used, via yaml files; or by inferring the build system in use by scanning the extracted directory. So for me it is not an issue. I am fully aware of differences from the user side; GNU configure is more convenient to use than the cmake options for example.)

**Comment by zarniwhoop73** [comment]
If the old configure system is not supported, there is little point me detailing the problem. And I'm not proposing to offer a patch, I test things, not develop them. So before I took time to investigate the differences between what configure supports and what cmake maybe supports, I figured I should ask if the configure script is just legacy.

I agree wholeheartedly that from the user viewpoint configure is usually easier - I've lost count of the number of weird differences between cmake projects.
